src/json.c

```diff
@@ -77,7 +83,79 @@ int FieldSpec_CheckJsonType(FieldType fieldType, JSONType type) {
   return rv;
 }
 
-int JSON_StoreInDocField(RedisJSON json, JSONType jsonType, struct DocumentField *df) {
+static int JSON_getInt32(RedisJSON json, int32_t *val) {
+  long long temp;
+  int ret = japi->getInt(json, &temp);
+  *val = (int32_t)temp;
+  return ret;
+}
+
+static int JSON_getFloat32(RedisJSON json, float *val) {
+  double temp;
+  int ret = japi->getDouble(json, &temp);
+  *val = (float)temp;
+  return ret;
+}
+
+int JSON_StoreVectorInDocField(FieldSpec *fs, JSONResultsIterator arrIter, struct DocumentField *df) {
+  VecSimType type = (VecSimType)0;
+  size_t dim = 0;
+  int (*getFunc)(RedisJSON, void *) = (int (*)(RedisJSON, void *))JSON_getFloat32;
```

Comment:
Currently these initializations are redundant since they are overwritten in all switch cases below, and the default assigned values could be wrong.

If not now, at least for later, seems better to fail if not all enum values are handled
(by default it is a warning: enumeration value %s not handled in switch [-Wswitch])

Adding `default` switch case, is not better than the initial initializations we have now, and it might even lower code coverage.

What about adding compilation flag, e.g.,  to VecSim/vec_sim_common.h
```
#pragma GCC diagnostic push
#pragma GCC diagnostic error "-Wswitch"
```
Which will enforce all enums to be handled, and there we would initialize accordingly, avoiding any "fake" defaults.

Reply:
I added this because a compiler warning about uninitialized variable. will this change fix that?